### DISCUSSION TITLE
Scarpet spawn_potential returns 0?

### DISCUSSION THREAD

**Discussion by RandomGgames** [gnembon/fabric-carpet#1348] [Q&A]
![image](https://user-images.githubusercontent.com/17207260/155085706-f9f745ff-41b9-49e3-bcc3-54d9db8e9574.png)

So when I place a pink carpet, it shows that mobs are capable of spawning at this location. However when running the command `/script run spawn_potential(pos(player()))` (without the carpet placed), it returns 0. Does this function not show the spawn potential of mobs?

If not, how could I scan a block and see if anything can spawn there (given can and fit are both true)? I'm basically trying to check if a block can spawn a mob above it, and if so, place a block to prevent it... A spawnproofing script that doesn't just place a block everywhere, but only where mobs could actually spawn.

**Comment by altrisi** [comment]
Spawn potential is a different thing. It's for the potential used in some nether mobs to determine whether they should spawn based on biome and other mobs around (for example endermen in warped forest, ghasts in sould sand valleys and I believe striders).

You could check the docs for something about mob spawning, not sure if there's anything, but if not you can always do `run('spawn ...')` and parse the command output to see if mobs can spawn (that's the system the pink carpet uses under the hood, it's basically a shortcut).

**Reply by RandomGgames** [reply]
Doesn't look like the spawn function cares... spawns the mob regardless of what the pink carpet says fits or not.
